### PR TITLE
Add more functions to fetch and url for sdk

### PR DESCRIPTION
**Pull request by karthik2804**
Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

### CODE REVIEW

File: crates/spin-js-engine/src/lib.rs [hunk 1]
```diff
@@ -103,10 +103,10 @@ fn spin_send_http_request(context: &Context, _this: &Value, args: &[Value]) -> R
                 .uri(request.uri.deref());
 
             if let Some(headers) = builder.headers_mut() {
-                for (key, value) in &request.headers {
+                for header in &request.headers {
```
Comment (dicej) [inline crates/spin-js-engine/src/lib.rs]: What's the reason for this change?

Comment (karthik2804) [reply crates/spin-js-engine/src/lib.rs]: This was an oversight on my part. It is unnecessary. I just rewrote that piece of code.

**Review by dicej — APPROVED** [review]
